src/libs/antares/study/parallel-years.cpp

```diff
@@ -0,0 +1,431 @@
+/*
+** Copyright 2007-2022 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include <utility>
+#include <vector>
+#include <climits>
+#include <cassert>
+
+#include <yuni/core/system/cpu.h> // For use of Yuni::System::CPU::Count()
+#include <yuni/yuni.h>
+#include <yuni/core/string.h>
+#include <yuni/core/fwd.h>
+
+#include <antares/logs.h>
+#include <antares/utils.hxx>
+#include "parameters.h"
+#include "parallel-years.h"
+
+#define SEP Yuni::IO::Separator
+
+namespace Antares
+{
+
+/*!
+** \brief Checks if clusters have the "gen-ts" param
+**
+** \warning TODO: REMOVE THIS METHOD ONCE THE TSGEN HAS BEEN DECOUPLED FROM SOLVER
+*/
+
+bool TempAreaListHolder::checkThermalTSGeneration(YString folder_)
```

Comment:
Thermal clusters are special regarding TS refresh. 
As for any other production mean, clusters have their own global parameter to enable time series refreshment. But user can ask the refresh to any particular cluster, that is not to obey to the global parameter.
Here we check if **any** cluster has to refresh its TS,  and we return true in that case, false otherwise.

**Question** : why get this information form input files, and not from memory ?
Probably because when we build the parallel years orchestrator, this information on clusters is not loaded yet.
In that case, why not build the parallel years orchestrator after the information on clusters is loaded ?